pandas/core/indexes/range.py

```diff
@@ -62,6 +61,37 @@
 _dtype_int64 = np.dtype(np.int64)
 
 
+def has_range_delta(values) -> bool | int:
+    """
+    Check if values have unique difference for RangeIndex._shallow_copy.
+
+    values must have more than 2 values.
+    If there is a unique diff, it cannot be zero.
+
+    Parameters
+    ----------
+    values : 1D iterable
+
+    Returns
+    -------
+    bool or int
+        False if there isn't a unique delta
+        int if there's a non-zero, unique delta
+    """
+    if len(values) < 2:
+        return False
+    diff = values[1] - values[0]
+    if diff == 0:
+        return False
+    curr = values[1]
```

Comment:
This is still pretty slow I think, I think what you are looking for is ``is_range_indexer`` that we built for CoW with a stepsize argument added

Reply:
Ah gotcha. I refactored to use `is_range_indexer`

Reply:
Just to be clear, that will only work with stepsize 1 for now, I am open to support other step sizes as well if you like (follow up should be fine)

Reply:
Right I "normalize" the values by the stepsize (`values - values[0] / values[1] - values[0]`) so that `is_range_indexer` can always be used here

Reply:
Ah sorry, yeah that makes sense